### DISCUSSION TITLE
4.7.25 Client host rejected: cannot find your hostname, [172.25.0.1]

### DISCUSSION THREAD

**Discussion by GoliathLabs** [dani-garcia/vaultwarden#2647] [Q&A]
### Subject of the issue

I've configured SMTP properly yet I can't deliver my mail (e.g. by testing the mail) as a valid hostname has to be sent to the mailserver.
### Deployment environment

* vaultwarden version: Latest (1.25.1)

* Install method: Docker image

* Clients used: 

* Reverse proxy and version: 
traefik 2.8.1
* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:
Relevant config:
```
  "_enable_smtp": true,
  "smtp_host": "smtp.mydomain.org",
  "smtp_security": "starttls",
  "smtp_port": 587,
  "smtp_from": "no-reply@mydomain.org",
  "smtp_from_name": "Vaultwarden",
  "smtp_username": "no-reply@mydomain.org",
  "smtp_password": "supersecretpasswordforgithub",
  "smtp_auth_mechanism": "LOGIN",
  "smtp_timeout": 15,
  "helo_name": "mail.mydomain.org",
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "_enable_email_2fa": false,
  "email_token_size": 6,
  "email_expiration_time": 600,
  "email_attempts_limit": 3
```

Mailserver log:
```
mailserver-mailserver-1  | Jul 26 22:02:47 mail postfix/submission/smtpd[4570]: connect from unknown[172.25.0.1]
mailserver-mailserver-1  | Jul 26 22:02:47 mail postfix/submission/smtpd[4570]: Anonymous TLS connection established from unknown[172.25.0.1]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest SHA256
mailserver-mailserver-1  | Jul 26 22:02:47 mail postfix/submission/smtpd[4570]: NOQUEUE: reject: RCPT from unknown[172.25.0.1]: 450 4.7.25 Client host rejected: cannot find your hostname, [172.25.0.1]; from=<no-reply@mydomain.org> to=<myname@mydomain.org> proto=ESMTP helo=<mail.mydomain.org>
mailserver-mailserver-1  | Jul 26 22:02:47 mail postfix/submission/smtpd[4570]: disconnect from unknown[172.25.0.1] ehlo=2 starttls=1 auth=1 mail=1 rcpt=0/1 quit=1 commands=6/7
```

### Steps to reproduce

See above

### Expected behaviour

The mail goes through as with my other services

### Actual behaviour

Didn't work as shown above

**Comment by BlackDex** [comment]
This has nothing to do with Vaultwarden, it looks like your mailserver can't find the hostname of the IP connecting.

**Reply by GoliathLabs** [reply]
Seems like it's a problem if the sending server (vaultwarden) and the mailserver share the same host. But I think you are right that it isn't vaultwarden related.

**Comment by Masgalor** [comment]
I may add some postfix-knowledge.
But first of all a little disclaimer `Do not change postfix rejects/permits just because some guy online told you to.`, some folks might not see how dangerous this is.

Your problem is caused by either [reject_unknown_client_hostname](https://www.postfix.org/postconf.5.html#reject_unknown_client_hostname) or [reject_unknown_reverse_client_hostname](https://www.postfix.org/postconf.5.html#reject_unknown_reverse_client_hostname) in your postfix config. Maybe there are other checks that trigger this error, but those are the common ones.

These checks are usually skipped for internal traffic by using [permit_inet_interfaces](https://www.postfix.org/postconf.5.html#permit_inet_interfaces) or [permit_mynetworks](https://www.postfix.org/postconf.5.html#permit_mynetworks) or [permit_sasl_authenticated](https://www.postfix.org/postconf.5.html#permit_sasl_authenticated) .

If I had to take a guess, I'd say `172.25.0.1` is missing from `mynetworks`.

**Reply by polarathene** [reply]
> But first of all a little disclaimer `Do not change postfix rejects/permits just because some guy online told you to.`, some folks might not see how dangerous this is.

Are you saying that kind of advice in general (_some guy online advising something_) is dangerous, or that specific quoted advice? Especially in context of the `reject_unknown_client_hostname` or `reject_unknown_reverse_client_hostname` being added as a restriction?

If specific to the Postfix restrictions, then sure it's helpful to understand the setting and impact of restrictions applied, along with their order.
